docs/adrs/00004-importer-scalability.md

```diff
@@ -0,0 +1,62 @@
+# 00004. Importer Scalability in Trustify
+
+Date: 2025-02-14
+
+## Status
+
+Draft
+
+## Context
+
+This ADR is concerned with our ability to scale the importer processes
+that ingest documents -- SBOM's and advisories -- from various
+sources.
+
+Currently, the importer process iterates over the enabled importer
+definitions in the database sequentially, running each according to
+its schedule. Executing more importer processes results in redundant
+work, each of them doing the same thing, i.e. there is no coordination
+to distribute the load among them.
+
+### Requirements
+
+* The work should be distributed -- no two importer processes should
+  be doing the same work.
+* The solution should be fault-tolerant -- importer instances may come
+  and go -- any work in process by a crashed/killed one must be
+  eventually picked up by another.
+* In short, adding an importer process should increase the ingestion
+  throughput, i.e. the system should be scalable.
+
+## Decision
+
+Since trustify already depends on PostgreSQL -- the import definitions
```

Comment:
If/when we have a user that wants to use some other database - I do not think we have 'painted ourselves in a corner' eg. there maybe some small work to get things running elsewhere ... to me this is a small theoretical risk.

Reply:
FYI, we do use a few pg-specific features in the query stuff, e.g. ILIKE, so something to stay mindful of.